packages/xod-client/src/editor/actions.js

```diff
@@ -717,71 +725,89 @@ export const runTabtest = patchPath => (dispatch, getState) => {
     ProjectSelectors.getProject
   )(getState());
 
+  const abortSymbol = Symbol('ABORT');
+  const shouldContinue = () => Selectors.isTabtestRunning(getState());
+  const abortOrPass = fn => x => {
+    if (!shouldContinue()) {
+      return Promise.reject(abortSymbol);
```

Comment:
A new pattern detected. I’d stick to the current system: a custom `code` of a new `Error` object